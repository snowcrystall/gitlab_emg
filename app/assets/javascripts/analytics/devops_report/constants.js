import { __ } from '~/locale';

export const INTRO_COOKIE_KEY = 'dev_ops_report_intro_callout_dismissed';

export const INTRO_BANNER_TITLE = __('Introducing Your DevOps Report');

export const INTRO_BANNER_BODY = __(
  'Your DevOps Report gives an overview of how you are using GitLab from a feature perspective. Use it to view how you compare with other organizations.',
);

export const INTRO_BANNER_ACTION_TEXT = __('Read more');
