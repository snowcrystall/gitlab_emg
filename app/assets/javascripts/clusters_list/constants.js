import { __, s__ } from '~/locale';

export const CLUSTER_ERRORS = {
  default: {
    tableText: s__('ClusterIntegration|Unknown Error'),
    title: s__('ClusterIntegration|Unknown Error'),
    description: s__(
      'ClusterIntegration|An unknown error occurred while attempting to connect to Kubernetes.',
    ),
    troubleshootingTips: [
      s__('ClusterIntegration|Check your cluster status'),
      s__('ClusterIntegration|Make sure your API endpoint is correct'),
      s__(
        'ClusterIntegration|Node calculations use the Kubernetes Metrics API. Make sure your cluster has metrics installed',
      ),
    ],
  },
  authentication_error: {
    tableText: s__('ClusterIntegration|Unable to Authenticate'),
    title: s__('ClusterIntegration|Authentication Error'),
    description: s__('ClusterIntegration|GitLab failed to authenticate.'),
    troubleshootingTips: [
      s__('ClusterIntegration|Check your token'),
      s__('ClusterIntegration|Check your CA certificate'),
    ],
  },
  connection_error: {
    tableText: s__('ClusterIntegration|Unable to Connect'),
    title: s__('ClusterIntegration|Connection Error'),
    description: s__('ClusterIntegration|GitLab failed to connect to the cluster.'),
    troubleshootingTips: [
      s__('ClusterIntegration|Check your cluster status'),
      s__('ClusterIntegration|Make sure your API endpoint is correct'),
    ],
  },
  http_error: {
    tableText: s__('ClusterIntegration|Unable to Connect'),
    title: s__('ClusterIntegration|HTTP Error'),
    description: s__('ClusterIntegration|There was an HTTP error when connecting to your cluster.'),
    troubleshootingTips: [s__('ClusterIntegration|Check your cluster status')],
  },
};

export const CLUSTER_TYPES = {
  project_type: __('Project'),
  group_type: __('Group'),
  instance_type: __('Instance'),
};

export const MAX_REQUESTS = 3;

export const STATUSES = {
  default: { className: 'bg-white', title: __('Unknown') },
  disabled: { className: 'disabled', title: __('Disabled') },
  created: { className: 'bg-success', title: __('Connected') },
  unreachable: { className: 'bg-danger', title: __('Unreachable') },
  authentication_failure: { className: 'bg-warning', title: __('Authentication Failure') },
  deleting: { title: __('Deleting') },
  creating: { title: __('Creating') },
};
