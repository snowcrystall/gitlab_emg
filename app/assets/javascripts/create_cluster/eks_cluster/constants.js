export const DEFAULT_REGION = 'us-east-2';

export const KUBERNETES_VERSIONS = [
  { name: '1.15', value: '1.15' },
  { name: '1.16', value: '1.16' },
  { name: '1.17', value: '1.17' },
  { name: '1.18', value: '1.18' },
  { name: '1.19', value: '1.19', default: true },
];
