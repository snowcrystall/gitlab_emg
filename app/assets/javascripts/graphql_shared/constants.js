export const TYPE_CI_RUNNER = 'Ci::Runner';
export const TYPE_EPIC = 'Epic';
export const TYPE_GROUP = 'Group';
export const TYPE_ISSUE = 'Issue';
export const TYPE_ITERATION = 'Iteration';
export const TYPE_ITERATIONS_CADENCE = 'Iterations::Cadence';
export const TYPE_MERGE_REQUEST = 'MergeRequest';
export const TYPE_MILESTONE = 'Milestone';
export const TYPE_PROJECT = 'Project';
export const TYPE_SCANNER_PROFILE = 'DastScannerProfile';
export const TYPE_SITE_PROFILE = 'DastSiteProfile';
export const TYPE_USER = 'User';
export const TYPE_VULNERABILITY = 'Vulnerability';
