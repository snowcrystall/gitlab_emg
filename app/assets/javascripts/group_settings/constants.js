import { __ } from '~/locale';

// Debounce delay in milliseconds
export const DEBOUNCE_TOGGLE_DELAY = 1000;

export const ERROR_MESSAGE = __('Refresh the page and try again.');

// runner setting options
export const ENABLED = 'enabled';
export const DISABLED = 'disabled_and_unoverridable';
export const ALLOW_OVERRIDE = 'disabled_with_override';
