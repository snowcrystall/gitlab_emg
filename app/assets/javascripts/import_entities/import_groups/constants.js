import { s__ } from '~/locale';

export const i18n = {
  NAME_ALREADY_EXISTS: s__('BulkImport|Name already exists.'),
};

export const NEW_NAME_FIELD = 'new_name';
