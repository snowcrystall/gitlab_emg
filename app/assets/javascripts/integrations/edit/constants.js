import { s__ } from '~/locale';

export const integrationLevels = {
  GROUP: 'group',
  INSTANCE: 'instance',
};

export const defaultIntegrationLevel = integrationLevels.INSTANCE;

export const overrideDropdownDescriptions = {
  [integrationLevels.GROUP]: s__(
    'Integrations|Default settings are inherited from the group level.',
  ),
  [integrationLevels.INSTANCE]: s__(
    'Integrations|Default settings are inherited from the instance level.',
  ),
};
