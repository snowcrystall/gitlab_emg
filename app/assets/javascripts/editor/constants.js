import { DEFAULT_DEBOUNCE_AND_THROTTLE_MS } from '~/lib/utils/constants';
import { __ } from '~/locale';

export const SOURCE_EDITOR_INSTANCE_ERROR_NO_EL = __(
  '"el" parameter is required for createInstance()',
);

export const URI_PREFIX = 'gitlab';
export const CONTENT_UPDATE_DEBOUNCE = DEFAULT_DEBOUNCE_AND_THROTTLE_MS;

export const ERROR_INSTANCE_REQUIRED_FOR_EXTENSION = __(
  'Source Editor instance is required to set up an extension.',
);

export const EDITOR_READY_EVENT = 'editor-ready';

export const EDITOR_TYPE_CODE = 'vs.editor.ICodeEditor';
export const EDITOR_TYPE_DIFF = 'vs.editor.IDiffEditor';

export const EDITOR_CODE_INSTANCE_FN = 'createInstance';
export const EDITOR_DIFF_INSTANCE_FN = 'createDiffInstance';

//
// EXTENSIONS' CONSTANTS
//

// For CI config schemas the filename must match
// '*.gitlab-ci.yml' regardless of project configuration.
// https://gitlab.com/gitlab-org/gitlab/-/issues/293641
export const EXTENSION_CI_SCHEMA_FILE_NAME_MATCH = '.gitlab-ci.yml';

export const EXTENSION_MARKDOWN_PREVIEW_PANEL_CLASS = 'md';
export const EXTENSION_MARKDOWN_PREVIEW_PANEL_PARENT_CLASS = 'source-editor-preview';
export const EXTENSION_MARKDOWN_PREVIEW_ACTION_ID = 'markdown-preview';
export const EXTENSION_MARKDOWN_PREVIEW_PANEL_WIDTH = 0.5; // 50% of the width
export const EXTENSION_MARKDOWN_PREVIEW_UPDATE_DELAY = 250; // ms
