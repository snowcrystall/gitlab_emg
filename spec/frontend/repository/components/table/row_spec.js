import { GlBadge, GlLink, GlIcon } from '@gitlab/ui';
import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import { createMockDirective, getBinding } from 'helpers/vue_mock_directive';
import TableRow from '~/repository/components/table/row.vue';
import FileIcon from '~/vue_shared/components/file_icon.vue';
import { FILE_SYMLINK_MODE } from '~/vue_shared/constants';

let vm;
let $router;

function factory(propsData = {}) {
  $router = {
    push: jest.fn(),
  };

  vm = shallowMount(TableRow, {
    propsData: {
      ...propsData,
      name: propsData.path,
      projectPath: 'gitlab-org/gitlab-ce',
      url: `https://test.com`,
      totalEntries: 10,
    },
    directives: {
      GlHoverLoad: createMockDirective(),
    },
    provide: {
      glFeatures: { refactorBlobViewer: true },
    },
    mocks: {
      $router,
    },
    stubs: {
      RouterLink: RouterLinkStub,
    },
  });

  vm.setData({ escapedRef: 'main' });
}

describe('Repository table row component', () => {
  const findRouterLink = () => vm.find(RouterLinkStub);

  afterEach(() => {
    vm.destroy();
  });

  it('renders table row', () => {
    factory({
      id: '1',
      sha: '123',
      path: 'test',
      type: 'file',
      currentPath: '/',
    });

    return vm.vm.$nextTick().then(() => {
      expect(vm.element).toMatchSnapshot();
    });
  });

  it('renders a symlink table row', () => {
    factory({
      id: '1',
      sha: '123',
      path: 'test',
      type: 'blob',
      currentPath: '/',
      mode: FILE_SYMLINK_MODE,
    });

    return vm.vm.$nextTick().then(() => {
      expect(vm.element).toMatchSnapshot();
    });
  });

  it('renders table row for path with special character', () => {
    factory({
      id: '1',
      sha: '123',
      path: 'test$/test',
      type: 'file',
      currentPath: 'test$',
    });

    return vm.vm.$nextTick().then(() => {
      expect(vm.element).toMatchSnapshot();
    });
  });

  it('renders a gl-hover-load directive', () => {
    factory({
      id: '1',
      sha: '123',
      path: 'test',
      type: 'blob',
      currentPath: '/',
    });

    const hoverLoadDirective = getBinding(findRouterLink().element, 'gl-hover-load');

    expect(hoverLoadDirective).not.toBeUndefined();
    expect(hoverLoadDirective.value).toBeInstanceOf(Function);
  });

  it.each`
    type        | component         | componentName
    ${'tree'}   | ${RouterLinkStub} | ${'RouterLink'}
    ${'blob'}   | ${RouterLinkStub} | ${'RouterLink'}
    ${'commit'} | ${'a'}            | ${'hyperlink'}
  `('renders a $componentName for type $type', ({ type, component }) => {
    factory({
      id: '1',
      sha: '123',
      path: 'test',
      type,
      currentPath: '/',
    });

    return vm.vm.$nextTick().then(() => {
      expect(vm.find(component).exists()).toBe(true);
    });
  });

  it.each`
    path
    ${'test#'}
    ${'Ã„nderungen'}
  `('renders link for $path', ({ path }) => {
    factory({
      id: '1',
      sha: '123',
      path,
      type: 'tree',
      currentPath: '/',
    });

    return vm.vm.$nextTick().then(() => {
      expect(vm.find({ ref: 'link' }).props('to')).toEqual({
        path: `/-/tree/main/${encodeURIComponent(path)}`,
      });
    });
  });

  it('renders link for directory with hash', () => {
    factory({
      id: '1',
      sha: '123',
      path: 'test#',
      type: 'tree',
      currentPath: '/',
    });

    return vm.vm.$nextTick().then(() => {
      expect(vm.find('.tree-item-link').props('to')).toEqual({ path: '/-/tree/main/test%23' });
    });
  });

  it('renders commit ID for submodule', () => {
    factory({
      id: '1',
      sha: '123',
      path: 'test',
      type: 'commit',
      currentPath: '/',
    });

    return vm.vm.$nextTick().then(() => {
      expect(vm.find('.commit-sha').text()).toContain('1');
    });
  });

  it('renders link with href', () => {
    factory({
      id: '1',
      sha: '123',
      path: 'test',
      type: 'blob',
      url: 'https://test.com',
      currentPath: '/',
    });

    return vm.vm.$nextTick().then(() => {
      expect(vm.find('a').attributes('href')).toEqual('https://test.com');
    });
  });

  it('renders LFS badge', () => {
    factory({
      id: '1',
      sha: '123',
      path: 'test',
      type: 'commit',
      currentPath: '/',
      lfsOid: '1',
    });

    return vm.vm.$nextTick().then(() => {
      expect(vm.find(GlBadge).exists()).toBe(true);
    });
  });

  it('renders commit and web links with href for submodule', () => {
    factory({
      id: '1',
      sha: '123',
      path: 'test',
      type: 'commit',
      url: 'https://test.com',
      submoduleTreeUrl: 'https://test.com/commit',
      currentPath: '/',
    });

    return vm.vm.$nextTick().then(() => {
      expect(vm.find('a').attributes('href')).toEqual('https://test.com');
      expect(vm.find(GlLink).attributes('href')).toEqual('https://test.com/commit');
    });
  });

  it('renders lock icon', () => {
    factory({
      id: '1',
      sha: '123',
      path: 'test',
      type: 'tree',
      currentPath: '/',
    });

    vm.setData({ commit: { lockLabel: 'Locked by Root', committedDate: '2019-01-01' } });

    return vm.vm.$nextTick().then(() => {
      expect(vm.find(GlIcon).exists()).toBe(true);
      expect(vm.find(GlIcon).props('name')).toBe('lock');
    });
  });

  it('renders loading icon when path is loading', () => {
    factory({
      id: '1',
      sha: '1',
      path: 'test',
      type: 'tree',
      currentPath: '/',
      loadingPath: 'test',
    });

    expect(vm.find(FileIcon).props('loading')).toBe(true);
  });
});
