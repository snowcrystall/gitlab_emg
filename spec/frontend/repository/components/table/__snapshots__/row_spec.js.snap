// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository table row component renders a symlink table row 1`] = `
<tr
  class="tree-item"
>
  <td
    class="tree-item-file-name cursor-default position-relative"
  >
    <a
      class="tree-item-link str-truncated"
      data-qa-selector="file_name_link"
      href="https://test.com"
      title="test"
    >
      <file-icon-stub
        class="mr-1 position-relative text-secondary"
        cssclasses="position-relative file-icon"
        filemode="120000"
        filename="test"
        size="16"
      />
      <span
        class="position-relative"
      >
        test
      </span>
    </a>
     
    <!---->
     
    <!---->
     
    <!---->
  </td>
   
  <td
    class="d-none d-sm-table-cell tree-commit cursor-default"
  >
    <gl-skeleton-loading-stub
      class="h-auto"
      lines="1"
    />
  </td>
   
  <td
    class="tree-time-ago text-right cursor-default"
  >
    <gl-skeleton-loading-stub
      class="ml-auto h-auto w-50"
      lines="1"
    />
  </td>
</tr>
`;

exports[`Repository table row component renders table row 1`] = `
<tr
  class="tree-item"
>
  <td
    class="tree-item-file-name cursor-default position-relative"
  >
    <a
      class="tree-item-link str-truncated"
      data-qa-selector="file_name_link"
      href="https://test.com"
      title="test"
    >
      <file-icon-stub
        class="mr-1 position-relative text-secondary"
        cssclasses="position-relative file-icon"
        filemode=""
        filename="test"
        size="16"
      />
      <span
        class="position-relative"
      >
        test
      </span>
    </a>
     
    <!---->
     
    <!---->
     
    <!---->
  </td>
   
  <td
    class="d-none d-sm-table-cell tree-commit cursor-default"
  >
    <gl-skeleton-loading-stub
      class="h-auto"
      lines="1"
    />
  </td>
   
  <td
    class="tree-time-ago text-right cursor-default"
  >
    <gl-skeleton-loading-stub
      class="ml-auto h-auto w-50"
      lines="1"
    />
  </td>
</tr>
`;

exports[`Repository table row component renders table row for path with special character 1`] = `
<tr
  class="tree-item"
>
  <td
    class="tree-item-file-name cursor-default position-relative"
  >
    <a
      class="tree-item-link str-truncated"
      data-qa-selector="file_name_link"
      href="https://test.com"
      title="test"
    >
      <file-icon-stub
        class="mr-1 position-relative text-secondary"
        cssclasses="position-relative file-icon"
        filemode=""
        filename="test"
        size="16"
      />
      <span
        class="position-relative"
      >
        test
      </span>
    </a>
     
    <!---->
     
    <!---->
     
    <!---->
  </td>
   
  <td
    class="d-none d-sm-table-cell tree-commit cursor-default"
  >
    <gl-skeleton-loading-stub
      class="h-auto"
      lines="1"
    />
  </td>
   
  <td
    class="tree-time-ago text-right cursor-default"
  >
    <gl-skeleton-loading-stub
      class="ml-auto h-auto w-50"
      lines="1"
    />
  </td>
</tr>
`;
